geoapi/services/projects.py

```diff
@@ -277,23 +226,25 @@ def getFeatures(projectId: int, query: dict = None) -> object:
         return out.geojson
 
     @staticmethod
-    def update(projectId: int, data: dict) -> Project:
+    def update(user: User, projectId: int, data: dict) -> Project:
         """
         Update the metadata associated with a project
         :param projectId: int
         :param data: dict
         :return: Project
         """
-        current_project = ProjectsService.get(project_id=projectId)
+        proj = ProjectsService.get(project_id=projectId)
+
+        for key, value in data['project'].items():
+            setattr(proj, key, value)
 
-        current_project.name = data['name']
-        if 'description' in data:
-            current_project.description = data['description']
-        if 'public' in data:
-            current_project.public = data['public']
         db_session.commit()
 
-        return current_project
+        if data.get('observable', False):
```

Comment:
update:  maybe add ticket for follow on (?) **but for now** just only allow users to update the name and description and not things like `observable` or the paths?

Reply:
Gotcha. I'll drop this for now. I think some code when I was trying to figure out save-as.
This could get solved when working on the save-as ticket https://jira.tacc.utexas.edu/browse/DES-2156.